client/galaxy/scripts/components/History/model/caching/schema/datasetCollection.schema.js

```diff
@@ -0,0 +1,81 @@
+import { nullableString, stringArray } from "./types.schema";
+
+export default {
+    title: "datasetCollections",
+    version: 0,
+    type: "object",
+    properties: {
+        history_content_type: { type: "string" },
+        populated_state_message: nullableString,
+        name: { type: "string" },
+        populated: { type: "boolean" },
+        isDeleted: { type: "boolean" },
+        history_id: { type: "string" },
+        tags: stringArray,
+        visible: { type: "boolean" },
+        job_source_id: nullableString,
+        job_source_type: nullableString,
+        elements: {
+            type: "array",
+            items: { $ref: "#/definitions/treenode" }
+        },
+        collection_type: { type: "string" },
+        url: { type: "string" },
+        model_class: nullableString,
+        hid: { type: "integer" },
+        element_count: { type: "integer" },
+        id: { type: "string", primary: true },
+        populated_state: { type: "string" },
+        update_time: { type: "string" },
+        purged: { type: "boolean" },
+        type_id: { type: "string" }
+    },
+    definitions: {
+        treenode: {
+            type: "object",
+            properties: {
+                element_index: { type: "integer" },
+                element_identifier: { type: "string" },
+                element_type: { type: "string" },
+                model_class: { type: "string" },
+                id: { type: "string" },
+
+                // find and kill the person who named this 'object'
```

Comment:
It is impressive code but you need to not threaten to kill me in it!

Reply:
Ooooh, so you confess!

Yeah I think I had a dyslexic stroke when I was wading through a tree structure with names like "element", and "object", haha.